src/_algopy_testing/arc4.py

```diff
@@ -179,12 +182,20 @@ def __hash__(self) -> int:
         return hash(self.bytes)
 
 
-def arc4_signature(signature: str, /) -> algopy.Bytes:
+def arc4_signature(signature: str | Callable[_P, _R], /) -> algopy.Bytes:
     """Convert a signature to ARC4 bytes."""
     import algopy
 
+    from _algopy_testing.decorators.arc4 import get_arc4_metadata
+
+    method_signature = (
```

Comment:
I think this would be better as an if-else statement, so we can see from simple line coverage if both branches are tested.

Reply:
It has been updated to use `if-else` statement.